modules/puppet_proxy_common/custom_validators.rb

```diff
@@ -0,0 +1,12 @@
+module ::Proxy::Puppet
+  class Validators
+    class UrlValidator < ::Proxy::PluginValidators::Base
+      def validate!(settings)
```

Comment:
This is missing a check for the `:if` predicate in the plugin description:

    return true unless evaluate_predicate(settings)